community/values/src/main/java/org/neo4j/values/Values.java

```diff
@@ -0,0 +1,354 @@
+    {
+        return new ShortValue( value );
+    }
+
+    public static Value byteValue( byte value )
+    {
+        return new ByteValue( value );
+    }
+
+    public static Value booleanValue( boolean value )
+    {
+        return new BooleanValue( value );
+    }
+
+    public static Value charValue( char value )
+    {
+        return new CharValue( value );
+    }
+
+    public static Value doubleValue( double value )
+    {
+        return new DoubleValue( value );
+    }
+
+    public static Value floatValue( float value )
+    {
+        return new FloatValue( value );
+    }
+
+    public static Value stringArray( String[] value )
+    {
+        return new StringArray.Direct( value );
+    }
+
+    public static Value byteArray( byte[] value )
+    {
+        return new ByteArray.Direct( value );
+    }
+
+    public static Value longArray( long[] value )
+    {
+        return new LongArray.Direct( value );
+    }
+
+    public static Value intArray( int[] value )
+    {
+        return new IntArray.Direct( value );
+    }
+
+    public static Value doubleArray( double[] value )
+    {
+        return new DoubleArray.Direct( value );
+    }
+
+    public static Value floatArray( float[] value )
+    {
+        return new FloatArray.Direct( value );
+    }
+
+    public static Value booleanArray( boolean[] value )
+    {
+        return new BooleanArray.Direct( value );
+    }
+
+    public static Value charArray( char[] value )
+    {
+        return new CharArray.Direct( value );
+    }
+
+    public static Value shortArray( short[] value )
+    {
+        return new ShortArray.Direct( value );
+    }
+
+    // BOXED FACTORY METHODS
+
+    public static Value of( Object value )
+    {
+        if ( value instanceof String )
+        {
+            return stringValue( (String) value );
+        }
+        if ( value instanceof Object[] )
+        {
+            return arrayValue( (Object[]) value );
+        }
+        if ( value instanceof Long )
+        {
+            return longValue( (Long) value );
+        }
+        if ( value instanceof Integer )
+        {
+            return intValue( (Integer) value );
+        }
+        if ( value instanceof Boolean )
+        {
+            return booleanValue( (Boolean) value );
+        }
+        if ( value instanceof Double )
+        {
+            return doubleValue( (Double) value );
+        }
+        if ( value instanceof Float )
+        {
+            return floatValue( (Float) value );
+        }
+        if ( value instanceof Short )
+        {
+            return shortValue( (Short) value );
+        }
+        if ( value instanceof Byte )
+        {
+            return byteValue( (Byte) value );
+        }
+        if ( value instanceof Character )
+        {
+            return charValue( (Character) value );
+        }
+        if ( value instanceof byte[] )
+        {
+            return byteArray( ((byte[]) value).clone() );
```

Comment:
We should document the fact that this method makes defensive copies of arrays (and document for the specific methods that they _don't_ make defensive copies).